table/tables/index.go

```diff
@@ -485,50 +498,89 @@ func (c *index) Exist(sc *stmtctx.StatementContext, txn kv.Transaction, indexedV
 		if err != nil {
 			return false, nil, err
 		}
-
-		var (
-			tempKey []byte
-			keyVer  byte
-		)
 		// If index current is in creating status and using ingest mode, we need first
 		// check key exist status in temp index.
-		key, tempKey, keyVer = GenTempIdxKeyByState(c.idxInfo, key)
-		if keyVer != TempIndexKeyTypeNone {
-			KeyExistInfo, h1, err1 := KeyExistInTempIndex(context.TODO(), txn, tempKey, distinct, h, c.tblInfo.IsCommonHandle)
-			if err1 != nil {
-				return false, nil, err
-			}
-			switch KeyExistInfo {
-			case KeyInTempIndexNotExist, KeyInTempIndexIsDeleted:
-				return false, nil, nil
-			case KeyInTempIndexConflict:
-				return true, h1, kv.ErrKeyExists
-			case KeyInTempIndexIsItself:
-				continue
-			}
+		key, tempKey, _ := GenTempIdxKeyByState(c.idxInfo, key)
+		if len(tempKey) > 0 {
+			key = tempKey
 		}
-
-		value, err := txn.Get(context.TODO(), key)
-		if kv.IsErrNotFound(err) {
-			return false, nil, nil
+		foundKey, dupHandle, err := FetchDuplicatedHandle(context.TODO(), key, distinct, txn, c.tblInfo.ID, c.tblInfo.IsCommonHandle)
+		if err != nil || !foundKey {
+			return false, nil, err
 		}
-		if err != nil {
+		if dupHandle != nil && !dupHandle.Equal(h) {
 			return false, nil, err
 		}
+		continue
+	}
+	return true, h, nil
+}
 
-		// For distinct index, the value of key is handle.
-		if distinct {
-			var handle kv.Handle
-			handle, err := tablecodec.DecodeHandleInUniqueIndexValue(value, c.tblInfo.IsCommonHandle)
+// FetchDuplicatedHandle is used to find the duplicated row's handle for a given unique index key.
+func FetchDuplicatedHandle(ctx context.Context, key kv.Key, distinct bool,
+	txn kv.Transaction, tableID int64, isCommon bool) (foundKey bool, dupHandle kv.Handle, err error) {
+	val, err := txn.Get(ctx, key)
+	if err != nil {
+		if kv.IsErrNotFound(err) {
+			return false, nil, nil
+		}
+		// Unexpected errors.
```

Comment:
Can we print an error log?

Reply:
The error is returned to the outer function. Is it still necessary to add a log?